inserter/inserter.go

```diff
@@ -207,35 +207,54 @@ func NewProtoconfInserter(protoconfRoot string, kvStore store.Store) *ProtoconfI
 
 }
 
-var ErrInsertionCompleted = errors.New("insertion completed")
-
-func (i *ProtoconfInserter) InsertConfigFile(configFile string) error {
+func ensureConfigMaterialized(configFile string) string {
 	if !strings.HasSuffix(configFile, consts.CompiledConfigExtension) {
 		configFile = configFile + consts.CompiledConfigPath
 	}
-	metadata, err := i.GatherMetadata(filepath.Join(i.rel, consts.CompiledConfigPath, configFile))
+	if !strings.HasPrefix(configFile, consts.CompiledConfigPath) {
+		configFile = filepath.Join(consts.CompiledConfigPath, configFile)
+	}
+	return configFile
+}
+
+var ErrInsertionCompleted = errors.New("insertion completed")
+
+func (i *ProtoconfInserter) InsertConfigFile(configFile string) error {
+	configFile = ensureConfigMaterialized(configFile)
+	metadata, err := i.GatherMetadata(filepath.Join(i.rel, configFile))
 	if err != nil {
 		return err
 	}
-	configName := strings.TrimSuffix(configFile, consts.CompiledConfigExtension)
+	configName := strings.TrimPrefix(strings.TrimSuffix(configFile, consts.CompiledConfigExtension), consts.CompiledConfigPath)
 
 	protoconfValue := &datatypes.ProtoconfValue{}
-	err = i.parser.ReadConfig(filepath.Join(i.protoconfRoot, consts.CompiledConfigPath, configFile), protoconfValue)
+	err = i.parser.ReadConfig(filepath.Join(i.protoconfRoot, configFile), protoconfValue)
 	if err != nil {
 		return err
 	}
 	protoconfValue.Metadata = metadata
 	return i.InsertConfig(configName, protoconfValue, metadata)
+}
+
+func (i *ProtoconfInserter) getKvStoreForConfig(protoconfValue *datatypes.ProtoconfValue) store.Store {
```

Comment:
What if KV is not configmaps?

Reply:
Currently, the config maps implementation is the only one which uses namespaces. So specifying a namespace in the `RolloutConfig` implies you want to use a config map store